code/modules/clothing/spacesuits/rig/modules/specific/ai_container.dm

```diff
@@ -175,41 +184,214 @@
 	update_verb_holder()
 	return
+
+/*
+ * The realm of cyberpunk, a Neural Jack-enabled IIS.
+ * Components unashamedly utilizing the groundwork of borers.
+ */
+/obj/item/ai_verbs/advanced/verb/assume_control()
+	set category = "Hardsuit"
+	set name = "Enable Direct Control"
+	set src in usr
+
+	if(!usr.loc || !usr.loc.loc || !istype(usr.loc.loc, /obj/item/rig_module))
+		to_chat(usr, SPAN_WARNING("You are not loaded into a hardsuit."))
+		return
+
+	var/obj/item/rig_module/ai_container/advanced/module = usr.loc.loc
+	if(!module.holder)
+		to_chat(usr, SPAN_WARNING("Your module is not installed in a hardsuit."))
+		return
+
+	module.jack_brain()
+
+/obj/item/rig_module/ai_container/advanced
+	name = "\improper AINI module"
+	desc = "An artificial intelligence neural integration module, suitable for most hardsuits."
+	description_fluff = "The unassuming device has no exterior markings, save for a network usage monitor and tiny warning label. On closer inspection, it reads, \"Warning: Device may cause damage if used without proper neural interface firmware.\" Something in your core urges you to heed that warning."
+	description_info = "Unless used by a robotic RIG pilot, or an organic one with a neural implant, the module will cause brain-damage on activation of its override mode."
+	icon_state = "AIIS"
+	var/controlling = FALSE
+
+	verb_holder = /obj/item/ai_verbs/advanced
+
+	var/last_defib = 1
+	var/mob/living/captive_brain/pilot_brain	
+	var/obj/item/weapon/shockpaddles/rig/paddles
+	var/imperfect_connection = FALSE	// Is the AI properly connected to the body? If N, they can't speak without sign-lang.
+	var/corpse_pilot = FALSE	// Is the AI controlling a corpse? If Y, kill the body again when they revert.
+
+/obj/item/rig_module/ai_container/advanced/Initialize()
+	. = ..()
+
+	paddles = new(src)
+
+/obj/item/rig_module/ai_container/advanced/removed()
+	revert()
+	..()
+
+/obj/item/rig_module/ai_container/advanced/process()
+	if(holder)
+		if((holder.offline || !holder.ai_override_enabled) && controlling)
+			revert()
+
+		if(controlling)
+			var/mob/living/carbon/human/H = holder.wearer
+
+			var/mob/living/simple_mob/animal/borer/Borer = H.has_brain_worms()
+			if(Borer)	// As funny as it would be to see a borer, AI, and human swap minds like musical chairs, let's not.
+				revert()
+				to_chat(integrated_ai, SPAN_DANGER("<span class='danger'>The neural jack signals a warning abruptly, before rapidly retracting.</span>"))
+				to_chat(H, SPAN_WARNING("You are shoved into consciousness as though you have been dropped into freezing water."))
+				return
+
+			var/obj/item/organ/internal/brain/Brain = H.internal_organs_by_name[O_BRAIN]
+			if(Brain)
+				if(Brain.robotic < ORGAN_ASSISTED)
+					H.add_modifier(/datum/modifier/mute, 30 SECONDS, suppress_failure = TRUE)
+
+			if(H.stat == DEAD && world.time > last_defib + 1 MINUTE && (H.getBruteLoss() + H.getFireLoss()) < (H.getMaxHealth() * 2 - H.getCloneLoss()))
```

Comment:
```suggestion
			if(H.stat == DEAD && !paddles.cooldown && !paddles.can_revive(H))
```
That... should be sufficient to handle this check?
`paddles.can_revive()` returns an error string on failure, null on success, so we check that it's falsey.
You might need a little bit more logic to handle the setToxLoss() bit below, but that feels... extremely strong in the edge case where the user died entirely to toxloss and it just blows it all away.

Reply:
Toxloss in robots is purely from rads and tick damage from organs, which would still be present, in addition to the presence of the damage caused by the risen_corpse check.

Toxloss in meats is theoretically associated with organ failure and blood toxicity, which are both things generally disregarded by forcing the muscles to function.

In either case, the cause of damage will still probably be present in the body to cause expiry again, such as chems, rads, or damaged/rotting organs. At the point this is _strong_ is the case where, feasibly, the short moments the puppeteer gets to repair the body (if they can) or move them to safety are enough to maintain function until help arrives, or get somewhere others _can_ help, respectively.